packages/flame/lib/src/events/flame_game_mixins/scale_dispatcher.dart

```diff
@@ -0,0 +1,356 @@
+    // triggered. This line will serve as a reference to compute scale
+    // and rotation data.
+    _lineAtFirstUpdate ??= _currentLine;
+
+    // Do we also need to recompute local focal point,
+    // local relative to what ?
+    return ScaleUpdateDetails(
+      focalPoint: _computeFocalPoint(details),
+      rotation: _computeRotationFactor(details),
+      scale: _computeScale(details),
+      verticalScale: _computeVerticalScale(details),
+      horizontalScale: _computeHorizontalScale(details),
+      pointerCount: details.pointerCount,
+      focalPointDelta: details.focalPointDelta,
+      sourceTimeStamp: details.sourceTimeStamp,
+    );
+  }
+
+  /// Compute the focal point of the scale gesture using the one pointer
+  /// focal point (which is just the position of the pointer itself) and
+  /// the position of the last pointer triggering a drag update.
+  Offset _computeFocalPoint(ScaleUpdateDetails details) {
+    if (lastDragUpdate != null) {
+      return details.focalPoint + lastDragUpdate!.globalPosition / 2.0;
+    } else {
+      return details.focalPoint;
+    }
+  }
+
+  /// Compute the rotation of the scale gesture using the initial
+  /// line formed between the two pointers that form the scale gesture,
+  /// and the subsequent lines they form as they move. The rotation factor
+  /// is just the angle in radian between the two lines.
+  double _computeRotationFactor(ScaleUpdateDetails details) {
+    if (lastDragUpdate == null ||
+        _lineAtFirstUpdate == null ||
+        _currentLine == null) {
+      return 0.0;
+    }
+
+    final fx = _lineAtFirstUpdate!.pointerStartLocation.dx;
+    final fy = _lineAtFirstUpdate!.pointerStartLocation.dy;
+    final sx = _lineAtFirstUpdate!.pointerEndLocation.dx;
+    final sy = _lineAtFirstUpdate!.pointerEndLocation.dy;
+
+    final nfx = _currentLine!.pointerStartLocation.dx;
+    final nfy = _currentLine!.pointerStartLocation.dy;
+    final nsx = _currentLine!.pointerEndLocation.dx;
+    final nsy = _currentLine!.pointerEndLocation.dy;
+
+    final angle1 = math.atan2(fy - sy, fx - sx);
+    final angle2 = math.atan2(nfy - nsy, nfx - nsx);
+
+    return angle2 - angle1;
+  }
+
+  /// Compute the scale of the scale gesture using the initial
+  /// line formed between the two pointers that form the scale gesture,
+  /// and the subsequent lines they form as they move. The scale factor
+  /// is just length of current line over length of initial line.
+  double _computeScale(ScaleUpdateDetails details) {
+    if (lastDragUpdate == null ||
+        _currentLine == null ||
+        _lineAtFirstUpdate == null) {
+      return 1.0;
+    }
+
+    final currentLineDistance = _currentLine!.pointerStartLocation
+        .toVector2()
+        .distanceTo(_currentLine!.pointerEndLocation.toVector2());
+
+    final firstLineDistance = _lineAtFirstUpdate!.pointerStartLocation
+        .toVector2()
+        .distanceTo(_lineAtFirstUpdate!.pointerEndLocation.toVector2());
```

Comment:
This method creates a lot of new vector2 objects, it should be possible to calculate this without creating any new objects.